jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java

```diff
@@ -750,7 +678,7 @@ private byte[] prefixToBytes(int prefix, int length)
             return result;
 
         // Sets the _prefix_ most significant bits to 1
-        result[index] = (byte)~((1 << (8 - prefix)) - 1);
+        result[index] = (byte)-(1 << (8 - prefix));
```

Comment:
Any reason why you changed the bit manipulation? It looks correct, but it's always hard to review without tests.

Reply:
To keep intellij happy